app/src/main/java/com/google/android/samples/socialite/widget/SocialiteWidgetReceiver.kt

```diff
@@ -0,0 +1,273 @@
+import androidx.glance.LocalContext
+import androidx.glance.action.clickable
+import androidx.glance.appwidget.GlanceAppWidget
+import androidx.glance.appwidget.GlanceAppWidgetManager
+import androidx.glance.appwidget.GlanceAppWidgetReceiver
+import androidx.glance.appwidget.ImageProvider
+import androidx.glance.appwidget.SizeMode
+import androidx.glance.appwidget.action.actionStartActivity
+import androidx.glance.appwidget.components.Scaffold
+import androidx.glance.appwidget.components.TitleBar
+import androidx.glance.appwidget.lazy.LazyColumn
+import androidx.glance.appwidget.provideContent
+import androidx.glance.appwidget.updateAll
+import androidx.glance.layout.Alignment
+import androidx.glance.layout.Box
+import androidx.glance.layout.Column
+import androidx.glance.layout.ContentScale
+import androidx.glance.layout.Row
+import androidx.glance.layout.fillMaxSize
+import androidx.glance.layout.fillMaxWidth
+import androidx.glance.layout.padding
+import androidx.glance.layout.size
+import androidx.glance.layout.wrapContentHeight
+import androidx.glance.text.FontFamily
+import androidx.glance.text.FontWeight
+import androidx.glance.text.Text
+import androidx.glance.text.TextStyle
+import coil.Coil
+import coil.request.CachePolicy
+import coil.request.ErrorResult
+import coil.request.ImageRequest
+import coil.request.SuccessResult
+import coil.transform.CircleCropTransformation
+import com.google.android.samples.socialite.MainActivity
+import com.google.android.samples.socialite.R
+import com.google.android.samples.socialite.SocialApp
+import com.google.android.samples.socialite.model.Contact
+import com.google.android.samples.socialite.widget.model.WidgetModel
+import com.google.android.samples.socialite.widget.model.WidgetModelRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+class SocialiteWidgetReceiver : GlanceAppWidgetReceiver() {
+
+    override val glanceAppWidget: GlanceAppWidget
+        get() = SocialiteWidget()
+
+    override fun onDeleted(context: Context, appWidgetIds: IntArray) {
+        super.onDeleted(context, appWidgetIds)
+        val repository = (context.applicationContext as SocialApp).repository
+        repository.delete(appWidgetIds)
+    }
+}
+
+class SocialiteWidget : GlanceAppWidget() {
+
+    override val sizeMode: SizeMode
+        get() = SizeMode.Exact
+
+    override suspend fun provideGlance(context: Context, glanceId: GlanceId) {
+        val repository = (context.applicationContext as SocialApp).repository
+        val glanceAppWidgetManager = GlanceAppWidgetManager(context)
+        val widgetId = glanceAppWidgetManager.getAppWidgetId(glanceId)
+        Log.i("SOCIAL", "$widgetId")
+        provideContent {
+            Content(widgetId, repository, context)
+        }
+    }
+
+    companion object {
+        fun update(context: Context, repository: WidgetModelRepository) {
```

Comment:
Doesn't look like `repository` is needed here?

Reply:
Fixed